### DISCUSSION TITLE
Question: ordering of fire-and-forget job processing

### DISCUSSION THREAD

**Discussion by pfosser** [jobrunr/jobrunr#993] [Q&A] [answered]
Hi,
is there a specific order in which fire-and-forget jobs are processed? Maybe the order in which they are enqueued?
T.I.A. for your reply

**Comment by Donatavict** [comment, marked as answer]
The fire-and-forget jobs are processed in a FIFO manner (first in or first enqueued / first out). If you want the jobs to be processed in a specific order, you can use a JobRunr Pro function - priority queues.

**Reply by pfosser** [reply]
Thank you @Donatavict , what about the following situations:

1. a job is being retried because it failed. Do the following jobs keep waiting the entire retry-fail phase?
2. A job definitively fails (never succeeds the retries). Only at this point the following jobs are processed?

**Reply by Donatavict** [reply]
@pfosser First question - no, second question - not applicable. 

On exception, the job is being retried several times using an `ExponentionalBackoffPolicy` (which is configurable). All the other jobs are being processed in between the waiting time for retrying the job. 

On top of that, it also depends on the amount of workers (Threads) you have configured. So if you have 10 workers configured, 10 jobs will be processed in parallel and at the same time.

If you want your other jobs to wait before the first job either succeeds or fails, you can use the workflow feature from JobRunr Pro.

**Reply by pfosser** [reply]
Thank you @Donatavict
